comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -161,60 +161,77 @@ func (rc rcClient) start() {
+
+		// If either of the values were previously set via RC, log the current values now that we've unset them
+		if mrfFailoverMetricsSource == model.SourceRC {
+			pkglog.Infof("Falling back to `multi_region_failover.failover_metrics: %t`", config.Datadog.GetBool("multi_region_failover.failover_metrics"))
+		}
+		if mrfFailoverLogsSource == model.SourceRC {
+			pkglog.Infof("Falling back to `multi_region_failover.failover_logs: %t`", config.Datadog.GetBool("multi_region_failover.failover_logs"))
+		}
+		return
+	}
+
 	applied := false
 	for cfgPath, update := range updates {
 		mrfUpdate, err := parseMultiRegionFailoverConfig(update.Config)
 		if err != nil {
-			pkglog.Errorf("MRF update unmarshal failed: %s", err)
+			pkglog.Errorf("Multi-region Failover update unmarshal failed: %s", err)
+			applyStateCallback(cfgPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: err.Error(),
+			})
 			continue
 		}
-		if mrfUpdate != nil && mrfUpdate.Failover != nil {
+
+		if mrfUpdate != nil && (mrfUpdate.FailoverMetrics != nil || mrfUpdate.FailoverLogs != nil) {
+			// If we've received multiple config files updating the failover settings, we should disregard all but the first update and log it, as this is unexpected
 			if applied {
-				pkglog.Infof("Multiple MRF updates received, disregarding update of `multi_region_failover.failover_metrics`/`multi_region_failover.failover_logs` to %t", *mrfUpdate.Failover)
+				pkglog.Infof("Multiple multi-region failover updates received, disregarding update of `multi_region_failover.failover_metrics` to %v and `multi_region_failover.failover_logs` to %v", mrfUpdate.FailoverMetrics, mrfUpdate.FailoverLogs)
+				applyStateCallback(cfgPath, state.ApplyStatus{
+					State: state.ApplyStateError,
+					Error: "Multiple multi-region failover updates received. Only the first was applied.",
+				})
 				continue
 			}
-			failover = mrfUpdate.Failover
-			pkglog.Infof("Setting `multi_region_failover.failover_metrics` and `multi_region_failover.failover_logs` to `%t` through Remote Configuration", *failover)
-
-			applyError := false
-			for _, setting := range []string{"multi_region_failover.failover_metrics", "multi_region_failover.failover_logs"} {
-				// Don't try to apply any further if we already encountered an error while applying it to a specific setting.
-				if !applyError {
-					err = rc.settingsComponent.SetRuntimeSetting(setting, *failover, model.SourceRC)
-					if err != nil {
-						pkglog.Errorf("MRF failover update failed: %s", err)
-						applyError = true
-						applyStateCallback(cfgPath, state.ApplyStatus{
-							State: state.ApplyStateError,
-							Error: err.Error(),
-						})
-					}
+
+			if mrfUpdate.FailoverMetrics != nil {
+				err = rc.applyMRFRuntimeSetting("multi_region_failover.failover_metrics", *mrfUpdate.FailoverMetrics, cfgPath, applyStateCallback)
+				if err != nil {
+					continue
 				}
```

Comment:
This is the pedant in me, but could we have an info-level log here that says something like...

> Received remote update for Multi-Region Failover configuration: enabling failover for metrics

where `enabling` is just coming from something like:

```go
state := "disabling"
if *mrfUpdate.FailoverMetrics {
    state = "enabling"
}
```

and then change the log message in `applyMRFRuntimeSetting` to a debug-level log so there's a raw indication, during debugging, of the precise setting name being updated.

Basically, we have a lot of logging for MRF already that is very opaque, and I want to shape it more so that it's useful from a customer perspective: they shouldn't need to know the specific configuration setting to know metrics/logs failover is enabled, we can just tell them in plain English.